### PR TITLE
quantum mixer implementation

### LINKED ISSUE #641: :wrench: Implement a quantum mixer :mage:
# :wrench: Implement a quantum mixer :mage: 

## Introduction

Lightwood is an AutoML library that aims at abstracting common procedures in the Machine Learning usual workflow (e.g. data preparation, feature engineering, model building, model analysis) with an approach that can be fully automated but easily customizable and extended thanks to its flexibility. 

One of the key tenets of Lightwood's philosophy is to engineer features by using _encoders_ that might transform columnar data into a useful descriptor. These are then fed into what is called a _mixer_: the actual ML model tasked with learning to predict the specified target. 

This approach is very flexible. Recently, we've come across the great [Qiskit SDK](https://qiskit.org/) for working with quantum computers, and thought it could be very cool to experiment with _hybrid quantum-classical neural networks_ as mixers! 

## Task

Implement a proof-of-concept `QuantumMixer` class that inherits from our `BaseMixer` class and uses a hybrid quantum neural network as the learning mechanism. We suggest closely following this [Qiskit tutorial](https://qiskit.org/textbook/ch-machine-learning/machine-learning-qiskit-pytorch.html) for an initial implementation.

A good way to measure success is to implement a unit test that uses the quantum mixer in a simple task. For example, you can use [this integration test](https://github.com/mindsdb/lightwood/blob/46ba63adc9037bda45f0c864e72ec1da4029512e/tests/integration/basic/test_airline.py#L8) as reference and opt for the [HDI](https://github.com/mindsdb/lightwood/blob/staging/tests/data/hdi.csv) toy dataset. Be sure to read the [custom mixer tutorial](https://lightwood.io/tutorials/custom_mixer/custom_mixer.html) so that you can force quantum mixer usage in the test.

> Note: mixer accuracy is not relevant at this time, future work can tackle it. Additionally, you can opt for testing with synthetic data if that's easier for you.

The core Lightwood team would love to be of assistance to any contributors interested in tackling this challenge. If at any point you have questions and/or want to discuss relevant details, feel free to post them here or in the community slack.

## Steps :male_detective: :female_detective: 

- Fork the Lightwood repository, checkout the `staging` branch and from it create a new one.
- Implement the `QuantumMixer` class, using all resources suggested above. Any files should be added inside the `lightwood/encoder/quantum` directory.
- Implement a unit test where your mixer is successfully used with a sample dataset.
- Make the PR and address any comments that reviewers might make.

## Additional rewards :1st_place_medal: 

This challenge encompasses mixer research. As such, an accepted PR brings :one::zero: points for entry into the draw for a :computer:  Deep Learning Laptop powered by the NVIDIA RTX 3080 Max-Q GPU or other swag :shirt: :bear: .  For more info check out https://mindsdb.com/hacktoberfest/

### PR DESCRIPTION
**Pull request by ongspxm**
closes #641 

implementing a quantum class neural network from https://qiskit.org/textbook/ch-machine-learning/machine-learning-qiskit-pytorch.html

## implementation notes
- currently using QClassic wrapper around the Neural mixer
  - its easier to error out the unit test
  - because qiskit is an optional dependency
- QClassicNet is quite rudimentary
  - returns a quantum modified value as per the tutorial

### CODE REVIEW

**Comment by ongspxm** [comment]
```
Running in /home/runner/work/lightwood/lightwood
Add input parsed as single string, running 1 git add command.
> Using 'Automated Author <info@mindsdb.com>' as author.
> Using "updating docs" as commit message.
> Running for a PR, the action will use 'feature/qmixer' as ref.
Internal logs
Outputs
Error: Error: There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> feature/qmixer
```

this github action seems to be failing, i ran the command locally but it doesnt seem to help resolve this issue, sorry new to this whole github action thing

**Comment by mrandri19** [comment]
(Author of the other PR here) I have to say that the two PRs came at around the same time. This one is more production-ready since it extends the pre-existing classes rather than creating new ones, hence taking advantage of much of the existing infrastructrue. If I had to choose, I would pick this.

I have a few comments, stemming from TODOs I have in my PR:
- Since this reuses the `Neural` class, I am wondering what will happen on a regression problem, especially considering the final softmax. Perhaps this should only support classification
- This only supports having quantum circuits as the last layer
- It would be cool to experiment with more complex quantum circuits

**Comment by ongspxm** [comment]
- yeah, you are right, softmax should be removed. in my mind, i saw crossentropyloss and i just put it in.
- added one last nit, for easy extension to other nets

yeah, i think it should be good to go
